### PR TITLE
Parse raw filter queries

### PR DESCRIPTION
**Pull request by manuelxaguilar**
### Summary:

When using a raw AWS FilterExpression on a scan, the results that are saved on the database as strings but that are declared on the schema as a different type (such as a prop of type Array with a default value of [] being stored as '[]') are not getting parsed before returning the response (as opposed to when you do a simple scan).
Added the already integrated parsing mechanism as a step for these cases.

### GitHub linked issue:

https://github.com/dynamoosejs/dynamoose/issues/550

### Type (select 1):
- [x] Bug fix
- [ ] Feature implementation
- [ ] Documentation improvement
- [ ] Testing improvement

- [ ] Test added to report bug (GitHub issue #--- @---)
- [ ] Something not listed here

### Is this a breaking change? (select 1):
- [ ] 🚨 YES 🚨
- [ ] No
- [x] I'm not sure

### Is this ready to be merged into Dynamoose? (select 1):
- [x] Yes
- [ ] No

### Are all the tests currently passing on this PR? (select 1):
- [x] Yes
- [ ] No

### Other:
- [x] I have searched through the GitHub pull requests to ensure this PR has not already been submitted
- [x] I have updated the Dynamoose documentation (if required) given the changes I made
- [x] I have added/updated the Dynamoose test cases (if required) given the changes I made
- [x] I have run `npm test` from the root of the project directory to ensure all tests continue to pass
- [x] I agree that all changes made in this pull request may be distributed and are made available in accordance with the [Dynamoose license](https://github.com/dynamoosejs/dynamoose/blob/master/LICENSE.txt)
- [x] All of my commits and commit messages are detailed, explain what changes were made, and are easy to follow and understand
- [x] I have confirmed that all my code changes are indented properly using 2 spaces
- [x] I have filled out all fields above

### CODE REVIEW

File: lib/Attribute.js [hunk 1]
```diff
@@ -556,7 +556,7 @@ Attribute.prototype.parseDynamo = async function(json) {
       return;
     }
     if(isSet) {
-      const set = json[type + 'S'];
+      const set = json[type + 'S'] || json;
```
Comment (fishcharlie) [inline lib/Attribute.js]: I'm concerned about this change. Isn't the Dynamoose JSON stored as something like the following?

```json
{
    "property": {
        "S": "Hello World"
    }
}
```

This change looks like it's assuming like it's the following.

```json
{
    "property": "Hello World"
}
```

File: lib/Scan.js [hunk 2]
```diff
@@ -82,20 +82,18 @@ Scan.prototype.exec = async function (next) {
...
 
           Object.keys(item).forEach(function (prop) {
             if (item[prop] instanceof DynamoDBSet) {
               item[prop] = item[prop].values;
             }
           });
 
-          model = new Model(item);
-          model.$__.isNew = false;
+          const model = await toModel(item)
```
Comment (fishcharlie) [inline lib/Scan.js]:
```suggestion
          const model = await toModel(item);
```

File: lib/Scan.js [hunk 3]
```diff
@@ -82,20 +82,18 @@ Scan.prototype.exec = async function (next) {
...
+        let returnItems = await Promise.all(data.Items.map(async function (item) {
 
           Object.keys(item).forEach(function (prop) {
             if (item[prop] instanceof DynamoDBSet) {
               item[prop] = item[prop].values;
             }
           });
 
-          model = new Model(item);
-          model.$__.isNew = false;
```
Comment (fishcharlie) [inline lib/Scan.js]: Why did this line get removed?

File: lib/Scan.js (hunk 2 above)
Comment (fishcharlie) [reply lib/Scan.js]: Mind also describing why this change was made? Or the reasoning behind it?

**Comment by comsky** [comment]
Hi there!
Is it still on progress? I'm stuck at the same issue.

My GraphQL query return type includes JSON format.
When I use scan with filter expression, it doesn't parse the result.

**1. scan.({Id: {eq: '12345'}}).exec()**
=> parsed result

**2. scan.({
FilterExpression: '#id = :val0',
ExpressionAttributeNames: { '#id': 'Id' }
ExpressionAttributeValues: { ':val0': '12345' }
}).exec()**
=> same result, but raw string format

**Pull request closed without merging**